packages/material-ui/src/Menu/Menu.js

```diff
@@ -29,92 +28,92 @@ export const styles = {
   },
 };
 
-class Menu extends React.Component {
-  menuListActionsRef = React.createRef();
+const Menu = React.forwardRef(function Menu(props, ref) {
+  const {
+    children,
+    classes,
+    disableAutoFocusItem,
+    MenuListProps,
+    onClose,
+    onEntering,
+    open,
+    PaperProps = {},
+    PopoverClasses,
+    theme,
+    ...other
+  } = props;
+  const menuListActions = React.useRef();
+
+  const focus = () => {
+    return menuListActions.current && menuListActions.current.focus();
+  };
 
-  componentDidMount() {
-    if (this.props.open && this.props.disableAutoFocusItem !== true) {
-      this.focus();
+  React.useEffect(() => {
+    if (open && !disableAutoFocusItem) {
+      focus();
     }
-  }
-
-  getContentAnchorEl = () => {
-    return this.menuListActionsRef.current.getContentAnchorEl();
-  };
+    // Needs to be called only on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
😕

Reply:
Why only on mount? Shouldn't it always focus when it opens? Or if not there should probably be an extra prop that determines if the menu should be focused when opened.

Reply:
Hmm, I was just copying what was there already. I guess it makes sense to have this when the props update.